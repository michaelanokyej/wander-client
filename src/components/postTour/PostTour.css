.div_form {
  width: 50%;
  position: relative;
  height: 50px;
  margin-top: 24px;
  overflow: hidden;
}

.div_form input {
  width: 100%;
  height: 100%;
  color: #595f6e;
  padding-top: 20px;
  border: none;
  outline: none;
}

.div_form label {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  pointer-events: none;
  border-bottom: 1px solid black;
}

.div_form label::after {
  content: "";
  position: absolute;
  left: 0px;
  bottom: -1px;
  height: 100%;
  width: 100%;
  border-bottom: 3px solid #13c776;
  transform: translateX(-100%);
  transition: transform all 0.3s ease;
}

.content-name {
  position: absolute;
  bottom: 5px;
  left: 0px;
  transition: all 0.3s ease;
  padding-left: 4px;
}

.div_form input:focus + .label-name .content-name,
.div_form input:valid + .label-name .content-name {
  transform: translateY(-120%);
  font-size: 12px;
  color: #13c776;
}

.div_form input:focus + .label-name::after,
.div_form input:valid + .label-name::after {
  transform: translateX(0%);
}

.post-tour-button {
  margin-top: 16px;
}

@media (max-width: 768px) {
  .div_form {
    width: 90%;
  }
}
