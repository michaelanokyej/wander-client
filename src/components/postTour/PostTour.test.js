import React from "react";
import PostTour from "./PostTour"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"

describe(`PostTour component`, () => {
    it(`renders the PostTour component`, () => {
        const wrapper = shallow(<PostTour />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});