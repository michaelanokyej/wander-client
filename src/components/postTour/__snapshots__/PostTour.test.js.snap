// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTour component renders the PostTour component 1`] = `
<main
  className="post-tour-body"
>
  <header
    role="banner"
  >
    <h2>
      Post a tour!
    </h2>
  </header>
  <form
    className="signup-form"
  >
    <div
      className="div_form"
    >
      <input
        id="name"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <label
        className="label-name"
        htmlFor="name"
      >
        <span
          className="content-name"
        >
          Tour Name
        </span>
      </label>
    </div>
    <div
      className="div_form"
    >
      <input
        id="city"
        name="city"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <label
        className="label-name"
        htmlFor="city"
      >
        <span
          className="content-name"
        >
          city
        </span>
      </label>
    </div>
    <div
      className="div_form"
    >
      <input
        id="state"
        maxLength="2"
        name="state"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <label
        className="label-name"
        htmlFor="state"
      >
        <span
          className="content-name"
        >
          State
        </span>
      </label>
    </div>
    <div
      className="div_form"
    >
      <input
        id="img"
        name="img"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <label
        className="label-name"
        htmlFor="img"
      >
        <span
          className="content-name"
        >
          Tour Image
        </span>
      </label>
    </div>
    <div
      className="div_form"
    >
      <input
        id="max_tourists"
        name="max_tourists"
        onChange={[Function]}
        required={true}
        type="number"
      />
      <label
        className="label-name"
        htmlFor="max_tourists"
      >
        <span
          className="content-name"
        >
          Max Capacity
        </span>
      </label>
    </div>
    <div
      className="div_form"
    >
      <input
        id="policies"
        name="policies"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <label
        className="label-name"
        htmlFor="policies"
      >
        <span
          className="content-name"
        >
          Policies
        </span>
      </label>
    </div>
    <div
      className="div_form"
    >
      <input
        id="description"
        name="description"
        onChange={[Function]}
        required={true}
      />
      <label
        className="label-name"
        htmlFor="description"
      >
        <span
          className="content-name"
        >
          Tour Description
        </span>
      </label>
    </div>
    <button
      className="post-tour-button"
      onClick={[Function]}
      type="submit"
    >
      Post tour
    </button>
  </form>
</main>
`;
